browser/ui/views/sidebar/sidebar_container_view.cc

```diff
@@ -791,98 +789,40 @@ void SidebarContainerView::OnEntryHidden(SidePanelEntry* entry) {
   }
 }
 
-void SidebarContainerView::OnEntryRegistered(SidePanelRegistry* registry,
-                                             SidePanelEntry* entry) {
-  // Observe when it's shown or hidden
-  DVLOG(1) << "Observing panel entry in registry observer: "
-           << SidePanelEntryIdToString(entry->key().id());
-  panel_entry_observations_.AddObservation(entry);
-}
-
-void SidebarContainerView::OnEntryWillDeregister(SidePanelRegistry* registry,
-                                                 SidePanelEntry* entry) {
-  // Stop observing
-  DVLOG(1) << "Unobserving panel entry in registry observer: "
-           << SidePanelEntryIdToString(entry->key().id());
-  panel_entry_observations_.RemoveObservation(entry);
-}
-
-void SidebarContainerView::OnRegistryDestroying(SidePanelRegistry* registry) {
-  if (panel_registry_observations_.IsObservingSource(registry)) {
-    StopObservingContextualSidePanelRegistry(registry);
-  }
+void SidebarContainerView::OnTabWillBeRemoved(content::WebContents* contents,
+                                              int index) {
+  // At this time, we can stop observing as TabFeatures is available.
+  StopObservingContextualSidePanelEntry(contents);
 }
 
 void SidebarContainerView::OnTabStripModelChanged(
     TabStripModel* tab_strip_model,
     const TabStripModelChange& change,
     const TabStripSelectionChange& selection) {
-  if ((change.type() == TabStripModelChange::kReplaced)) {
```

Comment:
@mkarolin To make this work w/o SidePanelRegistryObserver, I refactored more simply.
And I did test all sidebar scenarios I know but can you check also as you fixed bug recently?
Not sure this PR could be merged or not because upstream has regression after they deleted SidePanelRegistryObserver - https://chromium-review.googlesource.com/c/chromium/src/+/5884960/comment/9b1555b8_f9c16ec7/
but curious this PR works well.